src/VizPOIAnalytics/index.js

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import DeckGL from 'deck.gl';
+import { StaticMap } from 'react-map-gl';
+import POIAnalyticsControlPanel from './POIAnalyticsControlPanel';
+import POIAnalyticsRenderLayers from './POIAnalyticsRenderLayers';
+import * as CONSTANTS from './utils/constants';
+import lightingEffect from './utils/lightingEffects';
+import {
+  getPointCoords,
+  hexToDecimal,
+  getSumOfFoamTokens,
+  getValInUSD,
+} from './utils/helper';
+
+class VizPOIAnalytics extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getDataForCurrentViewport = this.getDataForCurrentViewport.bind(this);
+    this.state = {
+      viewport: {
+        longitude: -74,
+        latitude: 40.7,
+        zoom: 11,
+        minZoom: 5,
+        maxZoom: 16,
+        pitch: 45,
+        bearing: 0,
+      },
+      hover: {
+        x: 0,
+        y: 0,
+        hoveredObject: null,
+      },
+      points: [],
+      FOAMTokenInUSD: 0,
+      settings: Object.keys(CONSTANTS.HEXAGON_CONTROLS).reduce(
+        (accu, key) => ({
+          ...accu,
+          [key]: CONSTANTS.HEXAGON_CONTROLS[key].value,
+        }),
+        {},
+      ),
+    };
+  }
+
+  async componentDidMount() {
+    this.fetchAllPOIDetailsInCurrentViewport(CONSTANTS.boundingBoxNYC);
+    this.setUserLocation();
+    this.setState({
+      FOAMTokenInUSD: await getValInUSD(),
+    });
+  }
+
+  async onHover({ x, y, object }) {
+    const { FOAMTokenInUSD } = this.state;
+    if (object) {
+      const sumOfFoamTokens = object.points
+        ? await getSumOfFoamTokens(object.points)
+        : 0;
+      const details = {
+        latitude: object.position[0],
+        longitude: object.position[1],
+        numOfPoints: (object.points && object.points.length) || 0,
+        sumOfFoamTokens,
+        sumValInUSD: (sumOfFoamTokens * FOAMTokenInUSD).toFixed(2),
+      };
+      this.setState({
+        hover: {
+          x,
+          y,
+          hoveredObject: object,
+          details,
+        },
+      });
+    } else {
+      this.setState({ hover: { x, y, hoveredObject: object } });
+    }
+  }
+
+  setUserLocation() {
+    const { viewport } = this.state;
+    navigator.geolocation.getCurrentPosition((position) => {
+      const newViewport = {
+        ...viewport,
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+      };
+      this.setState({ viewport: newViewport });
+    });
+  }
+
+  getDataForCurrentViewport() {
+    const newBoundingBox = this.mapRef.getMap().getBounds();
+    const boundingBox = {
+      _ne: {
+        lng: newBoundingBox._ne.lng,
+        lat: newBoundingBox._ne.lat,
+      },
+      _sw: {
+        lng: newBoundingBox._sw.lng,
+        lat: newBoundingBox._sw.lat,
+      },
+    };
+    this.fetchAllPOIDetailsInCurrentViewport(boundingBox);
+  }
```

Comment:
So there are three different functions here: 

1. getBoundingBoxDetailsFromCurrentViewport which fetches the current viewPort. 
2. fetchPOIDetailsFromFOAMAPI which given bounding box fetches it from FOAMAPI
3. fetchAllPOIDetailsInCurrentViewport which encapsulates functionality from 1 and 2

Currently the code is confusing as it uses both getDataForCurrentViewport & fetchAllPOIDetailsInCurrentViewport which are actually just a single function. 


```
async fetchAllPOIDetailsInCurrentViewport() {

const boundingBoxDetailsFromCurrentViewPort = getBoundingBoxDetailsFromCurrentViewport(this.mapRef.getMap().getBounds())

const  pointsInCurrentViewPort = await fetchPOIDetailsFromFOAMAPI(boundingBoxDetailsFromCurrentViewPort)

this.setState({points: pointsInCurrentViewPort})
}

```

Remember this: One function must **do** only one thing. In the above `fetchAllPOIDetailsInCurrentViewport` the doing part is only setting state, rest is getting data for setting state.